ricecooker/chefs.py

```diff
@@ -281,10 +283,55 @@ def save_channel_tree_as_json(self, channel):
         self.CHEF_RUN_DATA['tree_archives']['current'] = filename.replace(os.getcwd() + '/', '')
         self.save_chef_data()
 
+    def save_channel_metadata_as_csv(self, channel):
+        # create data folder in chefdata
+        DATA_DIR = os.path.join('chefdata', 'data')
+        os.makedirs(DATA_DIR, exist_ok = True)
+        metadata_csv = csv.writer(open(os.path.join(DATA_DIR, 'content_metadata.csv'), 'w', newline='', encoding='utf-8'))
+        metadata_csv.writerow(config.CSV_HEADERS)
+
+        channel.save_channel_children_to_csv(metadata_csv)
+
+    def load_channel_metadata_from_csv(self):
+        metadata_dict = dict()
+        metadata_csv = None
+        CSV_FILE_PATH = os.path.join('chefdata', 'data', 'content_metadata.csv')
+        if os.path.exists(CSV_FILE_PATH):
+            metadata_csv = csv.DictReader(open(CSV_FILE_PATH, 'r', encoding='utf-8'))
+            for line in metadata_csv:
+                # Add to metadata_dict any updated data. Skip if none
+                line_source_id = line['Source ID']
+                line_new_title = line['New Title']
+                line_new_description = line['New Description']
+                line_new_tags = line['New Tags']
+                if line_new_title != '' or line_new_description != '' or line_new_tags != '':
+                    metadata_dict[line_source_id] = {}
+                    if line_new_title != '':
+                        metadata_dict[line_source_id]['New Title'] = line_new_title
+                    if line_new_description != '':
+                        metadata_dict[line_source_id]['New Description'] = line_new_description
+                    if line_new_tags != '':
+                        tags_arr = re.split(',| ,', line_new_tags)
+                        metadata_dict[line_source_id]['New Tags'] = tags_arr
+        return metadata_csv, metadata_dict
```

Comment:
We are still returning `metadata_csv` from this function and passing it into other functions, even though those functions do not access or use it.

Reply:
I am a little confused here. the `save_modifications` function does need `metadata_csv`. If no csv is passed in, then does not run but if there is a csv with any updated fields, then it does run.

Reply:
Yes, but what does it need the `metadata_csv` object for? It checks that it exists, but does nothing with it. If the `metadata_csv` file is `None`, then `metadata_dict` will also be `{}`, therefore the two checks are redundant, and you only need the `metadata_dict` check before trying to apply the modifications to the nodes.

Reply:
Thank you for the explanation, csv has been removed